File: app/Models/Post.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Post extends Model
{
    // Mendifiniskan field yang boleh di isi
    protected $fillable = ['title', 'content', 'category_id', 'petugas_id', 'status'];

    // Relasi Post ke Category (One to One)
    public function category()
    {
        return$this->belongsTo(Category::class);
    }

    // Relasi Post ke Petugas
    public function petugas(){
        return $this->belongsTo(Petugas::class);
    }

    // Relasi ke Gallery
    public function galleries()
    {
        return $this->hasMany(Gallery::class);
    }
}

File: lib/post.dart
class Post {
  final int id;
  final String title;
  final String content;
  final int categoryId;

  Post({required this.id, required this.title, required this.content, required this.categoryId});

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      title: json['title'] ?? 'No Title',
      content: json['content'] ?? 'No Content',
      categoryId: json['category_id'],
    );
  }
}
